### PR TITLE
[MLIR][TableGen] Minor code cleanup in DirectiveCommonGen

### PR DESCRIPTION
**Pull request by jurahul**
Directly Use Clause/ClauseVal as loop iterator.
Use llvm::transform instead of std::transform.
Use interleaveComma() to generate comma separated list.

### CODE REVIEW

**Review by clementval — COMMENTED** [review]
Looks like the pre ci is red. I don't know if this is related to this patch but please have a look.

File: mlir/tools/mlir-tblgen/DirectiveCommonGen.cpp [hunk 1]
```diff
@@ -84,11 +83,9 @@ static bool emitDecls(const RecordKeeper &recordKeeper, llvm::StringRef dialect,
     os << "  \"Clause" << enumName << "\",\n";
     os << "  \"" << enumName << " Clause\",\n";
     os << "  [";
-    for (unsigned int i = 0; i < cvDefs.size(); i++) {
-      os << cvDefs[i];
-      if (i != cvDefs.size() - 1)
-        os << ",";
-    }
+    llvm::ListSeparator LS;
+    for (const auto &cvDef : cvDefs)
```
[the remark this answers is not in the document]
Comment (jurahul) [reply mlir/tools/mlir-tblgen/DirectiveCommonGen.cpp]: Done.

**Comment by jurahul** [comment]
> Looks like the pre ci is red. I don't know if this is related to this patch but please have a look. 

Yes its unrelated and I've been trying to rebase to get it to clean state. It's clean now.

**Review by clementval — APPROVED** [review]
LGTM

**Pull request merged**